test/scenic_gen_component_test.exs

```diff
@@ -0,0 +1,66 @@
+Code.require_file("mix_helper.exs", __DIR__)
+
+defmodule Mix.Tasks.Scenic.Gen.ComponentTest do
+  use ExUnit.Case, async: false
+  @moduletag :gen_component
+
+  import ScenicNew.MixHelper
+  import ExUnit.CaptureIO
+
+  @component_name "test_component"
+  @component_module_name "TestComponent"
+  @app_name "scenic_demo"
+  @module_name "ScenicDemo"
+
+  test "gen.scene" do
+    Application.put_env(String.to_atom(@app_name), :namespace, String.to_atom(@module_name))
+
+    in_tmp("new with defaults", fn ->
+      func = fn ->
+        Mix.Tasks.Scenic.New.run([@app_name])
+        Mix.Tasks.Scenic.Gen.Component.run([@component_module_name])
+
+        assert_file("lib/components/#{@component_name}.ex", fn file ->
+          assert file =~ "defmodule #{@module_name}.Components.#{@component_module_name} do"
+        end)
+      end
+
+      assert capture_io(func) =~ "Created component #{@component_module_name}."
+    end)
+  end
```

Comment:
Just noticed that we can avoid one extra indention by simply moving `func` outside of `in_tmp/2` call:
```suggestion
  test "gen.scene" do
    Application.put_env(String.to_atom(@app_name), :namespace, String.to_atom(@module_name))

    func = fn ->
      Mix.Tasks.Scenic.New.run([@app_name])
      Mix.Tasks.Scenic.Gen.Component.run([@component_module_name])

      assert_file("lib/components/#{@component_name}.ex", fn file ->
        assert file =~ "defmodule #{@module_name}.Components.#{@component_module_name} do"
      end)
    end

    in_tmp("new with defaults", fn ->
      assert capture_io(func) =~ "Created component #{@component_module_name}."
    end)
  end
```